semcore/select/src/Select.jsx

```diff
@@ -81,20 +105,29 @@ class RootSelect extends Component {
       forwardRef,
       name,
       multiselect,
-      uid,
-      disablePortal,
       getI18nText,
+      highlightedIndex,
+      uid,
+      Children,
+      children: hasChildren,
     } = this.asProps;
 
+    const isMenu = isAdvanceMode(Children, [Select.Menu.displayName], true) || !hasChildren;
+    const ariaControls = isMenu ? `igc-${uid}-list` : `igc-${uid}-popper`;
+
     return {
-      id: `igc-${uid}-trigger`,
-      'aria-controls': visible ? `igc-${uid}-list` : undefined,
-      focusHint: visible && !disablePortal ? getI18nText('triggerHint') : undefined,
-      'aria-haspopup': 'listbox',
-      'aria-expanded': visible ? 'true' : 'false',
+      ...super.getTriggerProps(),
+      onKeyDown: callAllEventHandlers(
+        this.handlePreventCommonKeyDown.bind(this),
+        this.handleOpenKeyDown.bind(this),
+        this.handleArrowKeyDown.bind(this),
+      ),
+      'aria-controls': visible ? ariaControls : undefined,
+      'aria-haspopup': isMenu ? 'listbox' : 'dialog',
```

Comment:
I suggest making it always `aria-haspopup="listbox"`. 

We don't have examples with focusable elements other than the Search field, so in our examples users don't have to use Tab in Select.Popper. We'll add it as a recommendation, to change the value from "listbox" to "dialog" whenever they have something focusable other than the search field.

Reply:
especially since we now always have aria-activedescendant, regardles the popper structure, if I understand correctly

Reply:
That's not how it works. 
`dialog` is a default role of popper, but in `Select.Menu` we are `cancelling` it. 
So, we could cancel also role dialog in popper in `Select.Popper`, but I'm not sure that this is a right way.